graph-manager-docs/source/schema-registry.mdx

```diff
@@ -5,23 +5,71 @@ description: A central hub for your data graph
+Apollo Graph Manager's **schema registry** is a central hub for managing your data graph (as recommended by [Principled GraphQL](https://principledgraphql.com/integrity#3-track-the-schema-in-a-registry)). The schema registry enables you to explore your schema's types and fields, track its change history, and improve your production server's security by removing the need for introspection. You can also [register **variants** of your schema](#managing-environments-with-variants) that correspond to different environments, such as staging and production.
 
-<VideoFrame
-  title="Tracking your GraphQL Schema in a Registry"
-  src="https://www.youtube.com/embed/duwM95RsiBs"
-/>
+## Automatic registration
 
-## Prerequisites
+> **Automatic schema registration is not available for graphs that use Apollo Federation.** If you have a federated graph, instead see [Setting up managed federation](./managed-federation/setup/).
 
-To get started with the schema registry, first complete the following setup for the project that includes your GraphQL server:
+### Apollo Server setup
 
-<ProjectConfigPanel />
+Apollo Server version 2.15.1 and later can register its schema with Graph Manager automatically every time it starts up. To do so:
+
+1. Obtain an API key for your data graph:
+
+    <ObtainGraphApiKey />
+
+2. Set this API key as the value of the `APOLLO_KEY` environment variable in your server's environment.
+
+    If you're using the [`dotenv`](https://www.npmjs.com/package/dotenv) library, you can add the API key to the `.env` file in your project's root directory, like so:
+
+    ```none:title=.env
+    APOLLO_KEY=YOUR_KEY_HERE
+    ```
+
+3. Set the `reportSchema` option to `true` in your `ApolloServer` constructor, like so:
+
+    ```js
+    const server = new ApolloServer({
+      // ...other options...
+      reportSchema: true,
+    });
+    ```
+
+Now every time your server starts up, it registers its schema with Graph Manager.
 
-## Registering a schema manually
+#### Registering automatically to a variant
 
-Register your schema with Apollo Graph Manager by running the `apollo service:push` command from your application's root directory, like so:
+Your server can register its schema to a particular [variant of your graph](#managing-environments-with-variants). Each of your server's environments (development, staging, production, etc.) should register to a different variant.
+
+To do so, set the `APOLLO_GRAPH_VARIANT` environment variable in each server environment:
```

Comment:
Should we also mention you can set this in the `ApolloServer` constructor with the
```
engine: { graphVariant: "prod" }
```
option as well?

Reply:
I'd prefer to provide only a single method, and only the environment variable method. The constructor option method feels less appropriate for a value that changes specifically based on environment